src/dotty/tools/dotc/util/SourceFile.scala

```diff
@@ -36,9 +37,9 @@ object ScriptSourceFile {
 
 case class SourceFile(file: AbstractFile, content: Array[Char]) extends interfaces.SourceFile {
 
-  def this(_file: AbstractFile)                 = this(_file, _file.toCharArray)
-  def this(sourceName: String, cs: Seq[Char])   = this(new VirtualFile(sourceName), cs.toArray)
-  def this(file: AbstractFile, cs: Seq[Char])   = this(file, cs.toArray)
+  def this(_file: AbstractFile, codec: Codec) = this(_file, new String(_file.toByteArray, codec.charSet).toCharArray)
```

Comment:
I'm not sure using `Codec` is the best choice. `scala.io.*` is not really a nice API, and using it when we can do otherwise doesn't seem wise to me.

I suggest directly using `java.nio.Charset`.